Expose more structures and enums to IPC testing API

achristensen07 opened the pull request:
#### 1d5187ac0a4127da1c0600b9011f5f8563cd175b
<pre>
Expose more structures and enums to IPC testing API
<a href="https://bugs.webkit.org/show_bug.cgi?id=250957">https://bugs.webkit.org/show_bug.cgi?id=250957</a>
rdar://104518002

Reviewed by Simon Fraser.

* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUConvertToBackingContext.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUPowerPreference.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUTextureAspect.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUTextureFormat.h:
* Source/WebCore/bindings/js/BufferSource.h:
(WebCore::BufferSource::BufferSource):
(WebCore::BufferSource::span const):
(WebCore::BufferSource::encode const): Deleted.
(WebCore::BufferSource::decode): Deleted.
* Source/WebCore/platform/MediaSelectionOption.h:
(WebCore::MediaSelectionOption::encode const): Deleted.
(WebCore::MediaSelectionOption::decode): Deleted.
* Source/WebCore/platform/encryptedmedia/CDMEncryptionScheme.h:
* Source/WebCore/platform/encryptedmedia/CDMLogging.h:
* Source/WebCore/platform/encryptedmedia/CDMMediaCapability.h:
(WebCore::CDMMediaCapability::encode const): Deleted.
(WebCore::CDMMediaCapability::decode): Deleted.
* Source/WebCore/platform/encryptedmedia/CDMSessionType.h:
* Source/WebCore/platform/mediacapabilities/MediaDecodingType.h:
* Source/WebCore/platform/mediacapabilities/MediaEncodingType.h:
* Source/WebCore/platform/mediastream/MediaConstraints.h:
(WebCore::MediaTrackConstraintSetMap::encode const): Deleted.
(WebCore::MediaTrackConstraintSetMap::decode): Deleted.
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
(headers_for_type):
* Source/WebKit/Shared/PALArgumentCoders.serialization.in: Added.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/VisitedLinkStore.messages.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/VisitedLinkTableController.messages.in:

Canonical link: <a href="https://commits.webkit.org/259180@main">https://commits.webkit.org/259180@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/05a8b73fac5a680e96bd64165047db8ded5b7ba9

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104250 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13338 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37161 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113461 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/173750 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108181 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14416 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4251 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/96474 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112513 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110019 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/11098 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94166 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38765 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/107734 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/92958 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25777 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80427 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6699 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27137 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6833 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/3690 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12848 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46692 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6322 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8619 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Source/WebKit/WebProcess/WebPage/VisitedLinkTableController.messages.in:
```diff
@@ -22,7 +22,7 @@
 messages -> VisitedLinkTableController {
     SetVisitedLinkTable(WebKit::SharedMemory::Handle handle)
-    VisitedLinkStateChanged(Vector<WebCore::SharedStringHash> linkHashes)
+    VisitedLinkStateChanged(Vector<uint32_t> linkHashes)
```
This seems like a step backwards.

achristensen07 replied:
I don't completely disagree, but the C++ still has the more descriptive name.

achristensen07 committed:
Expose more structures and enums to IPC testing API
https://bugs.webkit.org/show_bug.cgi?id=250957
rdar://104518002

Reviewed by Simon Fraser.

* Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUConvertToBackingContext.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUPowerPreference.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUTextureAspect.h:
* Source/WebCore/PAL/pal/graphics/WebGPU/WebGPUTextureFormat.h:
* Source/WebCore/bindings/js/BufferSource.h:
(WebCore::BufferSource::BufferSource):
(WebCore::BufferSource::span const):
(WebCore::BufferSource::encode const): Deleted.
(WebCore::BufferSource::decode): Deleted.
* Source/WebCore/platform/MediaSelectionOption.h:
(WebCore::MediaSelectionOption::encode const): Deleted.
(WebCore::MediaSelectionOption::decode): Deleted.
* Source/WebCore/platform/encryptedmedia/CDMEncryptionScheme.h:
* Source/WebCore/platform/encryptedmedia/CDMLogging.h:
* Source/WebCore/platform/encryptedmedia/CDMMediaCapability.h:
(WebCore::CDMMediaCapability::encode const): Deleted.
(WebCore::CDMMediaCapability::decode): Deleted.
* Source/WebCore/platform/encryptedmedia/CDMSessionType.h:
* Source/WebCore/platform/mediacapabilities/MediaDecodingType.h:
* Source/WebCore/platform/mediacapabilities/MediaEncodingType.h:
* Source/WebCore/platform/mediastream/MediaConstraints.h:
(WebCore::MediaTrackConstraintSetMap::encode const): Deleted.
(WebCore::MediaTrackConstraintSetMap::decode): Deleted.
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/webkit/messages.py:
(serialized_identifiers):
(headers_for_type):
* Source/WebKit/Shared/PALArgumentCoders.serialization.in: Added.
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Source/WebKit/UIProcess/VisitedLinkStore.messages.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/WebPage/VisitedLinkTableController.messages.in:

Canonical link: https://commits.webkit.org/259180@main
